golang/testsuite/testsuite_impl/basic_datastore_and_api_test/basic_datastore_and_api_test_.go

```diff
@@ -6,49 +6,63 @@
 package basic_datastore_and_api_test
 
 import (
+	"encoding/json"
+	"fmt"
 	"github.com/kurtosis-tech/example-microservice/api/api_service_client"
 	"github.com/kurtosis-tech/example-microservice/datastore/datastore_service_client"
 	"github.com/kurtosis-tech/kurtosis-client/golang/networks"
 	"github.com/kurtosis-tech/kurtosis-client/golang/services"
 	"github.com/kurtosis-tech/kurtosis-libs/golang/lib/testsuite"
-	"github.com/kurtosis-tech/kurtosis-libs/golang/testsuite/services_impl/api"
-	"github.com/kurtosis-tech/kurtosis-libs/golang/testsuite/services_impl/datastore"
 	"github.com/palantir/stacktrace"
 	"github.com/sirupsen/logrus"
+	"os"
 )
 
 const (
-	datastoreServiceId services.ServiceID = "datastore"
-	apiServiceId services.ServiceID = "api"
-
-	waitForStartupDelayMilliseconds = 1000
-	waitForStartupMaxPolls = 15
-
-	testPersonId = 23
+	datastoreServiceId              services.ServiceID = "datastore"
+	apiServiceId                    services.ServiceID = "api"
+	datastorePort                                      = 1323
+	apiServicePort                                     = 2434
+	waitForStartupDelayMilliseconds                    = 1000
+	waitForStartupMaxPolls                             = 15
+
+	testPersonId     = 23
 	testNumBooksRead = 3
+
+	configFileKey = "config-file"
 )
 
 type BasicDatastoreAndApiTest struct {
-	datstoreImage string
-	apiImage string
+	datastoreImage string
+	apiImage       string
 }
 
-func NewBasicDatastoreAndApiTest(datstoreImage string, apiImage string) *BasicDatastoreAndApiTest {
-	return &BasicDatastoreAndApiTest{datstoreImage: datstoreImage, apiImage: apiImage}
+func NewBasicDatastoreAndApiTest(datastoreImage string, apiImage string) *BasicDatastoreAndApiTest {
+	return &BasicDatastoreAndApiTest{datastoreImage: datastoreImage, apiImage: apiImage}
 }
 
 func (b BasicDatastoreAndApiTest) Configure(builder *testsuite.TestConfigurationBuilder) {
 	builder.WithSetupTimeoutSeconds(60).WithRunTimeoutSeconds(60)
 }
 
 func (b BasicDatastoreAndApiTest) Setup(networkCtx *networks.NetworkContext) (networks.Network, error) {
-	datastoreConfigFactory := datastore.NewDatastoreContainerConfigFactory(b.datstoreImage)
-	datastoreServiceContext, datastoreSvcHostPortBindings, err := networkCtx.AddService(datastoreServiceId, datastoreConfigFactory)
+
+	containerCreationConfig := services.NewContainerCreationConfigBuilder(
```

Comment:
Recommend extracting this, and all other containerCreationConfig/runCreationConfigFuncs for all of these into private helper functions. We should still leave them as inside each test (let's not create an equivalent of ContainerCreationConfigFactory), but let's at least make them privat ehelper functions inside each test for clarity